fsspec/implementations/github.py

```diff
@@ -73,13 +78,16 @@ def repos(cls, org_or_user, is_org=True):
             Name of the github org or user to query
         is_org: bool (default True)
             Whether the name is an organisation (True) or user (False)
+        timeout: tuple of int
+            Connection and read timeouts, in seconds
 
         Returns
         -------
         List of string
         """
         r = requests.get(
-            f"https://api.github.com/{['users', 'orgs'][is_org]}/{org_or_user}/repos"
+            f"https://api.github.com/{['users', 'orgs'][is_org]}/{org_or_user}/repos",
+            timeout=timeout,
```

Comment:
Why not reuse `self.timeout` here?

Reply:
This is a `classmethod` and so `self.timeout` won't necessarily exist, right?

Reply:
I see what you mean. There should probably be a default value as a class attribute.

Reply:
I can do that. Just to be clear: I will set the `timeout` as a class attribute rather than as an instance attribute. This will mean (amongst other things) that if a GitHub filesystem instance changes the timeout value, this will also change the timeout for all other GitHub filesystem instances. Is this okay?

Reply:
> a class attribute rather than as an instance attribute

Both! If the user provides a non-None value, set it on the instance, so that `fs.timeout` will get the instance one if it exists but fall back to the class one otherwise.

Reply:
Done!